From a0f179673460e6377f1ca17b4ae51c8fe4c78d81 Mon Sep 17 00:00:00 2001
From: Yi Qingliang <niqingliang2003@gmail.com>
Date: Sat, 28 Sep 2019 19:21:21 +0800

---
 src/quick/items/qquicklistview.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/quick/items/qquicklistview.cpp b/src/quick/items/qquicklistview.cpp
index 0d78070401..f3511df542 100644
--- a/src/quick/items/qquicklistview.cpp
+++ b/src/quick/items/qquicklistview.cpp
@@ -3508,7 +3508,7 @@ bool QQuickListViewPrivate::applyInsertionChange(const QQmlChangeSet::Change &ch
             // Special case of appending an item to the model.
             index = visibleItems.count();
         } else {
-            if (modelIndex < visibleIndex) {
+            if (modelIndex <= visibleIndex) {
                 // Insert before visible items
                 visibleIndex += count;
                 for (FxViewItem *item : qAsConst(visibleItems)) {
@@ -3539,13 +3539,13 @@ bool QQuickListViewPrivate::applyInsertionChange(const QQmlChangeSet::Change &ch
     }
 
     bool visibleAffected = false;
-    if (insertResult->visiblePos.isValid() && pos < insertResult->visiblePos) {
+    if (insertResult->visiblePos.isValid() && pos <= insertResult->visiblePos) {
         // Insert items before the visible item.
         int insertionIdx = index;
         int i = 0;
         qreal from = tempPos - displayMarginBeginning - buffer;
 
-        if (insertionIdx < visibleIndex) {
+        if (insertionIdx <= visibleIndex) {
             if (pos >= from) {
                 // items won't be visible, just note the size for repositioning
                 insertResult->sizeChangesBeforeVisiblePos += count * (averageSize + spacing);
