From 3c7f41055c39eb494aa6fd39ff2c4f12bbb43789 Mon Sep 17 00:00:00 2001
From: Yi Qingliang <niqingliang2003@gmail.com>
Date: Tue, 1 Oct 2019 14:29:43 +0800
Subject: [PATCH] compile uboot for fsref

---
 arch/arm/dts/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 9fb38682e6..f6b117e6ef 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -295,6 +295,7 @@ dtb-$(CONFIG_ARCH_ZYNQ) += \
 	zynq-zturn.dtb \
 	zynq-zturn-v5.dtb \
 	zynq-zybo.dtb \
+	zynq-ax7021.dtb \
 	zynq-zybo-z7.dtb
 dtb-$(CONFIG_ARCH_ZYNQMP) += \
 	avnet-ultra96-rev1.dtb			\
