Index: libwebsockets-2.4.1/lib/libwebsockets.h
===================================================================
--- libwebsockets-2.4.1.orig/lib/libwebsockets.h
+++ libwebsockets-2.4.1/lib/libwebsockets.h
@@ -4013,9 +4013,6 @@ lws_set_timeout(struct lws *wsi, enum pe
 #if !defined(LWS_SIZEOFPTR)
 #define LWS_SIZEOFPTR (sizeof (void *))
 #endif
-#if !defined(u_int64_t)
-#define u_int64_t unsigned long long
-#endif
 
 #if defined(__x86_64__)
 #define _LWS_PAD_SIZE 16	/* Intel recommended for best performance */
